sstables/sstables_manager.cc

```diff
@@ -169,6 +169,27 @@ void sstables_manager::increment_total_reclaimable_memory_and_maybe_reclaim(ssta
     smlogger.info("Reclaimed {} bytes of memory from SSTable components. Total memory reclaimed so far is {} bytes", memory_reclaimed, _total_memory_reclaimed);
 }
 
+future<> sstables_manager::maybe_reload_reclaimed_components() {
+    size_t memory_reclaim_threshold = _available_memory * _db_config.components_memory_reclaim_threshold();
+    size_t memory_available_for_reclaimable_components = memory_reclaim_threshold - _total_reclaimable_memory;
+
+    while (memory_available_for_reclaimable_components > 0) {
+        auto sstable_to_reload = std::lower_bound(_reclaimed.begin(), _reclaimed.end(), memory_available_for_reclaimable_components, [](const sstable& sst, size_t memory_available_for_reclaimable_components) {
```

Comment:
I think this currently will try to reload the largest filters. I think we should reverse the sorting so we are loading the smallest filters first: this will maximize value, as we want to have the most filters in memory possible: so on evict we evict the largest (to minimize the amount of filters to evict) and on reload we reload the smallest (to maximize the amount of filters reloaded).